### DISCUSSION TITLE
Gatsby Wordpress ACF won't exposed fields if not used

### DISCUSSION THREAD

**Discussion by amirmeimari** [gatsbyjs/gatsby#26877] [Help]
This is so simple yet so frustrating!
I created a custom field in ACF and try to fetch data using `gatsby-source-wordpress` plugin in my blog posts.
If I don't use that custom field in any of my blog posts(in Wordpress admin panel when i try to publish or edit a post), gatsby graphql schema won't create that filed and I cannot query to get data. is there any way to edit graphql schema that comes from this plugin?
I want to be able to query this field if exists, if not it should return `null`. but field should be there in the schema in the first place otherwise build would fail!

My query: 
```
  query {
      allWordpressPost {
        edges {
          node {
            slug
            content
            acf { # because myCustomField is not avaible so this is not available too
              myCustomField # This is not available to query!
            }
          }
        }
      }
    }
```

The error I get:
```
There was an error in your GraphQL query:

Cannot query field "myCustomField " on type "wordpress__POSTAcf".

If you don't expect "myCustomField" to exist on the type "wordpress__POSTAcf" it is most likely a typo.
```

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

You'll want to use [Schema Customization](https://www.gatsbyjs.com/docs/schema-customization) to explicitly define the `myCustomField` - when it doesn't exist it'll return null.

I'd also encourage you to try out https://github.com/gatsbyjs/gatsby-source-wordpress-experimental as it'll also fix your issue with that, see: https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/blob/master/docs/problems-with-v3.md

> This means Gatsby needed to use inference to automatically build GraphQL types for the data that was returned from WP. The big issue with this is: if all data of a type is removed from WP, your Gatsby site's GraphQL queries will break because no data means Gatsby can no longer can infer the data structure. This led to a lot of odd and counterintuitive issues and a poor developer/content creator experience.

**Comment by amirmeimari** [comment]
Thanks, I'm currently trying v4 but I faced some [issues](https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/issues/172). Overall very good update 👍
